### DISCUSSION TITLE
Clone PVC in other namespaces

### DISCUSSION THREAD

**Discussion by dren-dk** [ceph/ceph-csi#2769] [Q&A] [answered]
# Describe the feature you'd like to have #

As far as I can tell there's no way to clone a PVC from another namespace, this would be quite handy in the situation where I'd like to clone a database from the production to the QA namespace. 

It seems there are various non-standard ways of doing this by other CSI drivers, such as setting a list of allowed downstream namespaces in an annotation of the source PVC and then pointing at the source PVC via an annotation of the clone PVC.

# What is the value to the end user? (why is it a priority?) #

Cloning PVCs within the same namespace only is severely limiting when you want to have production and test instances live in separate namespaces.

**Comment by Madhu-1** [comment]
@dren-dk what is the expectation from cephcsi here?

**Comment by dren-dk** [comment]
Ideally a pointer to documentation that says I'm mistaken and that there is a perfectly standard way to do what I want and that cephcsi supports it.

Secondarily I'd be happy with a non-standard cephcsi specific way to clone RBD PVCs across namespace boundaries.

**Comment by Madhu-1** [comment]
cephcsi is not blocking any clones across kubernetes namespaces. if there are any non-standard ways of doing it, the same should work for cephcsi also. 

AFAIK this is completely related to kubernetes and has nothing to do with cephcsi.

**Comment by Madhu-1** [comment, marked as answer]
You can try the below steps if you want to clone in a different namespace (just one hack of doing it)

* Create a PVC in one namespace
* Create clone is the same namespace
* Edit the PV/PVC reclaim policy and change it to Retain
* Delete the PVC and PVC reference in the PV object.
* Create the PVC in the different namespace and point to the above PV object.

**Comment by dren-dk** [comment]
Hmm, yes I guess that would be a useful workaround, though it relies on a lot of sequential steps rather than a static desired state, so it's harder to implement in a gitops workflow.

I'll go and have more of a think about this, thank you for taking the time to respond.

**Comment by Madhu-1** [comment]
Yes, it's hard but that's the one way of doing it. as nothing can be done from cephcsi for this one. please feel to reach out if you have any queries.
